### PR TITLE
Switch from k8s.gcr.io to registry.k8s.io

### PR DESCRIPTION
**Pull request by megian**
Reference: https://github.com/kubernetes/registry.k8s.io#background

### CODE REVIEW

File: release-tools/SIDECAR_RELEASE_PROCESS.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ naming convention `<hostpath-deployment-version>-on-<kubernetes-version>`.
 1. If release was a new major/minor version, create a new `release-<minor>`
    branch at that commit.
 1. Check [image build status](https://k8s-testgrid.appspot.com/sig-storage-image-build).
-1. Promote images from k8s-staging-sig-storage to k8s.gcr.io/sig-storage. From
+1. Promote images from k8s-staging-sig-storage to registry.k8s.io/sig-storage. From
```
Comment (pohly) [inline release-tools/SIDECAR_RELEASE_PROCESS.md:103]: Remove this change. "release-tools" is read-only in this repo, such a change needs to go into csi-release-tools.

Comment (megian) [reply release-tools/SIDECAR_RELEASE_PROCESS.md]: Removed.

**Comment by pohly** [comment]
/assign @wongma7

For approval.

**Comment by megian** [comment]
@wongma7 What's the current state of the PR approval?

File: charts/nfs-subdir-external-provisioner/values.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ replicaCount: 1
 strategyType: Recreate
 
 image:
-  repository: k8s.gcr.io/sig-storage/nfs-subdir-external-provisioner
+  repository: registry.k8s.io/sig-storage/nfs-subdir-external-provisioner
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/values.yaml:5]: please update the default value in the chart readme

https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/blame/master/charts/nfs-subdir-external-provisioner/README.md#L55

**Review by yonatankahana — CHANGES_REQUESTED** [review]

**Pull request merged**